### PR TITLE
refactor(router): update api contract for list payment methods to include connector

### PR DESCRIPTION
**Pull request by vspecky**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

This PR updates the contract of the List Payment Methods API to also include the connector name.

### Additional Changes
None
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Required by the frontend for showing the correct payment methods for the correct connector.

## How did you test it?

Local API testing, compiler-guided.
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by manoj-juspay — APPROVED** [review]
LGTM

**Review by SanchithHegde — APPROVED** [review]

File: crates/api_models/src/payment_methods.rs [hunk 1]
```diff
@@ -277,6 +277,12 @@ fn set_or_reject_duplicate<T, E: de::Error>(
     }
 }
 
+#[derive(Debug, serde::Serialize, ToSchema)]
+pub struct ListPaymentMethodConnectorObject {
+    pub connector_name: String,
```
Comment (jarnura) [inline crates/api_models/src/payment_methods.rs:282]: Add comments for each fields

File: crates/api_models/src/payment_methods.rs [hunk 2]
```diff
@@ -277,6 +277,12 @@ fn set_or_reject_duplicate<T, E: de::Error>(
     }
 }
 
+#[derive(Debug, serde::Serialize, ToSchema)]
+pub struct ListPaymentMethodConnectorObject {
```
Comment (jarnura) [inline crates/api_models/src/payment_methods.rs:281]: Object is a redundant word, avoid struct name with Object as suffix.

**Review by jarnura — CHANGES_REQUESTED** [review]

**Comment by vspecky** [comment]
Closing. Needs a different implementation and requires merging in release candidate branch.

**Pull request closed without merging**